src/neo4j_graphrag/experimental/pipeline/kg_builder.py

```diff
@@ -120,133 +86,29 @@ def __init__(
-        )
-        self.prompt_template = config.prompt_template
-        self.perform_entity_resolution = config.perform_entity_resolution
-        self.lexical_graph_config = config.lexical_graph_config
-        self.neo4j_database = config.neo4j_database
-
-        self.pipeline = self._build_pipeline()
-
-    @staticmethod
-    def to_schema_entity(entity: EntityInputType) -> SchemaEntity:
-        if isinstance(entity, dict):
-            return SchemaEntity.model_validate(entity)
-        return SchemaEntity(label=entity)
-
-    @staticmethod
-    def to_schema_relation(relation: RelationInputType) -> SchemaRelation:
-        if isinstance(relation, dict):
-            return SchemaRelation.model_validate(relation)
-        return SchemaRelation(label=relation)
-
-    def _build_pipeline(self) -> Pipeline:
-        pipe = Pipeline()
-
-        pipe.add_component(self.text_splitter, "splitter")
-        pipe.add_component(SchemaBuilder(), "schema")
-        pipe.add_component(
-            LLMEntityRelationExtractor(
-                llm=self.llm,
-                on_error=self.on_error,
-                prompt_template=self.prompt_template,
-            ),
-            "extractor",
-        )
-        pipe.add_component(TextChunkEmbedder(embedder=self.embedder), "chunk_embedder")
-        pipe.add_component(self.kg_writer, "writer")
-
-        if self.from_pdf:
-            pipe.add_component(self.pdf_loader, "pdf_loader")
-
-            pipe.connect(
-                "pdf_loader",
-                "splitter",
-                input_config={"text": "pdf_loader.text"},
-            )
-
-            pipe.connect(
-                "schema",
-                "extractor",
-                input_config={
-                    "schema": "schema",
-                    "document_info": "pdf_loader.document_info",
-                },
-            )
-        else:
-            pipe.connect(
-                "schema",
-                "extractor",
-                input_config={
-                    "schema": "schema",
-                },
-            )
-
-        pipe.connect(
-            "splitter", "chunk_embedder", input_config={"text_chunks": "splitter"}
-        )
-
-        pipe.connect(
-            "chunk_embedder", "extractor", input_config={"chunks": "chunk_embedder"}
-        )
-
-        # Connect extractor to writer
-        pipe.connect(
-            "extractor",
-            "writer",
-            input_config={"graph": "extractor"},
-        )
-
-        if self.perform_entity_resolution:
-            pipe.add_component(
-                SinglePropertyExactMatchResolver(
-                    self.driver, neo4j_database=self.neo4j_database
-                ),
-                "resolver",
+            config = SimpleKGPipelineConfig(
+                # argument type are fixed in the Config object
+                llm_config=llm,  # type: ignore[arg-type]
+                neo4j_config=driver,  # type: ignore[arg-type]
```

Comment:
These `type: ignore[arg-type]` can't be avoided?

Reply:
We could, but in that case that complexifies the code IMHO. The expected type is:

```
{"default": LLMType(llm)}
```

That's why we have validators in the `AbstractPipelineConfig` that transform the input to dict if it's not already one. The normal pydantic validation will introduce the `LLMType`.